src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellSectionRootHeader.cs

```diff
@@ -200,6 +200,10 @@ public override void ViewDidLoad()
 
 			UpdateSelectedIndex();
 			ShellSection.PropertyChanged += OnShellSectionPropertyChanged;
+			foreach (var shellContent in ShellSectionController.GetItems())
```

Comment:
This isn't going to work when items are added or removed.

When an item is removed or added `OnShellSectionItemsChanged` is called which calls `ReloadData` at which point I think your event subscription here will be out of sync.

Reply:
I have modified the code to address the ShellSection items changes. Please review and let me know